corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -426,12 +437,19 @@ def _diff_ledgers(self, case_ids):
                 filter_ledger_diffs(diffs)
             )
 
-    def _assert_no_migration_restrictions(self, domain_name):
-        assert should_use_sql_backend(domain_name)
-        assert not COUCH_SQL_MIGRATION_BLACKLIST.enabled(domain_name, NAMESPACE_DOMAIN)
-        assert not any(custom_report_domain == domain_name
-                       for custom_report_domain in settings.DOMAIN_MODULE_MAP.keys())
-        assert not REMINDERS_MIGRATION_IN_PROGRESS.enabled(domain_name)
+    def _check_for_migration_restrictions(self, domain_name):
+        if not should_use_sql_backend(domain_name):
+            msg = "does not have SQL backend not enabled"
```

Comment:
nit this is a double negative, but I think you meant `does not have SQL backend enabled`

Reply:
https://github.com/dimagi/commcare-hq/pull/24061/commits/850c7777d133e86a5b1d70f9badb2181d9e4d65a